soc/riscv/riscv-ite/it8xxx2/soc.c

```diff
@@ -191,8 +191,13 @@ void riscv_idle(enum chip_pll_mode mode, unsigned int key)
 	chip_pll_ctrl(mode);
 
 	do {
-		/* Wait for interrupt */
-		__asm__ volatile ("wfi");
+#ifdef CONFIG_I2C_IT8XXX2_CQ_MODE
+		if (!i2c_enhance_cpu_idle_not_allowed())
```

Comment:
There are different power saving states in the EC including Doze(Idle), Deep Doze, and the Sleep modes. The EC processor(CPU) cannot be in the Idle/Doze mode during the transactions with the CQ mode(DMA mode). Otherwise, the EC processor would be clock gated. The functions pm_policy_state_lock_get(), pm_policy_state_lock_put() can only stop the EC processor from entering the Deep Doze, but it’s still possible that EC processor enters the Doze(Idle) mode.

Reply:
Done.